Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,244 @@
 #include "simbody/internal/GeneralForceSubsystem.h"
 #include "simbody/internal/SimbodyMatterSubsystem.h"
 #include "simbody/internal/MultibodySystem.h"
+#include <iostream>
+#include <exception>
+#include <mutex>
+#include <thread>
 
 #include "ForceImpl.h"
 
+#include <memory>
+
+#define NUMNONPARALLELTHREADS 1
+#define NONPARALLELFORCESINDEX 0
```

Comment:
Don't use macros for constants in C++; used typed constants instead. If you want file-scope constants, use an anonymous namespace like this:
```c++
namespace {
    const int NumNonParallelThreads = 1;
    const int NonParallelForcesIndex = 0;
}
```
or you could use an enum.

General rule: macros are BAD!